src/standalone/Soil/energy_hydrology.jl

```diff
@@ -144,18 +146,21 @@ end
         domain::D,
         boundary_conditions::NamedTuple,
         sources::Tuple,
+        lateral_flow::Bool = true
     ) where {FT, D, PS}
 
-A constructor for a `EnergyHydrology` model.
+A constructor for a `EnergyHydrology` model, which sets the default value
+of the `lateral_flow` flag to true.
 """
 function EnergyHydrology{FT}(;
     parameters::EnergyHydrologyParameters{FT, PSE},
     domain::D,
     boundary_conditions::NamedTuple,
     sources::Tuple,
+    lateral_flow::Bool = true,
 ) where {FT, D, PSE}
     args = (parameters, domain, boundary_conditions, sources)
-    EnergyHydrology{FT, typeof.(args)...}(args...)
+    EnergyHydrology{FT, typeof.(args)...}(args..., lateral_flow)
```

Comment:
Could we add `lateral_flow` as the last item of `args` instead of passing it in separately? That might be a cleaner implementation

Reply:
hmmm we could but then lateral_flow is not part of the parametric types of the model, so we couldnt do {typeof.(args)...} here then: `EnergyHydrology{FT, typeof.(args)...}`

Reply:
(I declared lateral_flow as type Bool in the original struct defn)

Reply:
Ahhh I figured I was missing something. This looks good then :)